Makefile

```diff
@@ -561,9 +561,16 @@ else
 BUILD_DATE := 2020-01-01-00:00:00
 endif
 
-$(shell echo '#define QMK_VERSION "$(GIT_VERSION)"' > $(ROOT_DIR)/quantum/version.h)
-$(shell echo '#define QMK_BUILDDATE "$(BUILD_DATE)"' >> $(ROOT_DIR)/quantum/version.h)
-$(shell echo '#define CHIBIOS_VERSION "$(CHIBIOS_VERSION)"' >> $(ROOT_DIR)/quantum/version.h)
-$(shell echo '#define CHIBIOS_CONTRIB_VERSION "$(CHIBIOS_CONTRIB_VERSION)"' >> $(ROOT_DIR)/quantum/version.h)
+$(shell echo '#ifndef SKIP_VERSION' > $(ROOT_DIR)/quantum/version.h)
+$(shell echo '#  define QMK_VERSION "$(GIT_VERSION)"' >> $(ROOT_DIR)/quantum/version.h)
+$(shell echo '#  define QMK_BUILDDATE "$(BUILD_DATE)"' >> $(ROOT_DIR)/quantum/version.h)
+$(shell echo '#  define CHIBIOS_VERSION "$(CHIBIOS_VERSION)"' >> $(ROOT_DIR)/quantum/version.h)
+$(shell echo '#  define CHIBIOS_CONTRIB_VERSION "$(CHIBIOS_CONTRIB_VERSION)"' >> $(ROOT_DIR)/quantum/version.h)
+$(shell echo '#else' >> $(ROOT_DIR)/quantum/version.h)
```

Comment:
This is kinda duplicating the existing logic on lines 553-562. Why not just force SKIP_GIT too?

Reply:
`$(TOP)/Makefile` can not check the 'check-md5' target. Therefore, I made quantum/version.h as follows.

```c
#ifndef SKIP_VERSION
#  define QMK_VERSION "0.12.14-1-g32e4c2"
#  define QMK_BUILDDATE "2021-03-07-06:01:04"
#  define CHIBIOS_VERSION "breaking_2021_q1"
#  define CHIBIOS_CONTRIB_VERSION "breaking_2021_q1"
#else
#  define QMK_VERSION "NA"
#  define QMK_BUILDDATE "2020-01-01-00:00:00"
#  define CHIBIOS_VERSION "NA"
#  define CHIBIOS_CONTRIB_VERSION "NA"
#endif
```

And I made `$(TOP)/build_keyboard.mk` as follows.
```makefile
ifneq ($(filter check-md5,$(strip $(MAKECMDGOALS))),)
   SKIP_VERSION = yes
endif

ifdef SKIP_VERSION
    OPT_DEFS += -DSKIP_VERSION
endif
```

If there is a smarter way, I welcome your suggestions.